pkg/operator/cassandra/controller/util/util.go

```diff
@@ -0,0 +1,311 @@
+				"it is owned by '%v', not '%s/%s'.",
+			obj.GetNamespace(), obj.GetName(),
+			ownerRef,
+			owner.GetNamespace(), owner.GetName(),
+		)
+	}
+	return nil
+}
+
+// NewControllerRef returns an OwnerReference to
+// the provided Cluster Object
+func NewControllerRef(c *cassandrav1alpha1.Cluster) metav1.OwnerReference {
+	return *metav1.NewControllerRef(c, schema.GroupVersionKind{
+		Group:   cassandrarookio.GroupName,
+		Version: "v1alpha1",
+		Kind:    "Cluster",
+	})
+}
+
+// RefFromString is a helper function that takes a string
+// and outputs a reference to that string.
+// Useful for initializing a string pointer from a literal.
+func RefFromString(s string) *string {
+	return &s
+}
+
+func StatefulSetNameForRack(rackName, dcName, clusterName string) string {
+	return fmt.Sprintf("%s-%s-%s", clusterName, dcName, rackName)
+}
+
+func ServiceAccountNameForMembers(c *cassandrav1alpha1.Cluster) string {
+	return fmt.Sprintf("%s-member", c.Name)
+}
+
+func RoleNameForMembers(c *cassandrav1alpha1.Cluster) string {
+	return fmt.Sprintf("%s-member", c.Name)
+}
+
+func RoleBindingNameForMembers(c *cassandrav1alpha1.Cluster) string {
+	return fmt.Sprintf("%s-member", c.Name)
+}
+
+func HeadlessServiceNameForCluster(c *cassandrav1alpha1.Cluster) string {
+	return fmt.Sprintf("%s-hs", c.Name)
+}
+
+func CassandraImageForCluster(c *cassandrav1alpha1.Cluster) string {
+	repo := "cassandra"
+	if c.Spec.Repository != nil {
+		repo = *c.Spec.Repository
+	}
+	return fmt.Sprintf("%s:%s", repo, c.Spec.Version)
+}
+
+func StatefulSetForRack(r *cassandrav1alpha1.RackSpec, c *cassandrav1alpha1.Cluster, rookImage string) *appsv1.StatefulSet {
+
+	rackLabels := RackLabels(r.Name, c)
+	stsName := StatefulSetNameForRack(r.Name, c.Spec.Datacenter.Name, c.Name)
+
+	return &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            stsName,
+			Namespace:       c.Namespace,
+			Labels:          rackLabels,
+			OwnerReferences: []metav1.OwnerReference{NewControllerRef(c)},
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: &r.Members,
+			// Use a common Headless Service for all StatefulSets
+			ServiceName: HeadlessServiceNameForCluster(c),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: RackLabels(r.Name, c),
+			},
+			PodManagementPolicy: appsv1.OrderedReadyPodManagement,
+			UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+				Type: appsv1.RollingUpdateStatefulSetStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: rackLabels,
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "shared",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:            "cassandra",
+							Image:           CassandraImageForCluster(c),
+							ImagePullPolicy: "IfNotPresent",
+							Ports: []corev1.ContainerPort{
```

Comment:
these container ports look familiar, if they are repeated, could the be refactored to a common place/function?